app/models/judgement.rb

```diff
@@ -9,35 +9,45 @@ class Judgement < ApplicationRecord
 	# Generating CSV for all judgements
 	def self.to_csv
 		CSV.generate do |csv|
+			
+			finalKeyArr = Array.new
+			keyArr = Judgement.first.attributes.keys
+			customScoreArr = Judgement.first.attributes.values[6]
 
-			@header = []
-			Judgement.attribute_names.each do |attr_name|
-				if attr_name == 'user_id'
-					@header << 'Judge Name'
-				elsif attr_name == 'project_id'
-					@header << 'Project Name'
-				elsif attr_name == 'custom_scores'
-					# TODO: Improve Display of Custom Scores
-					@header << 'Custom Scores'
-				else
-					@header << attr_name.titleize
+			for i in 0..keyArr.length
+				if(keyArr[i] != "custom_scores" && keyArr[i] != " ")
+					finalKeyArr.push(keyArr[i])
 				end
 			end
-			csv << @header
 
+			finalKeyArr.delete_at(finalKeyArr.length() - 1)
+			
+			for key, value in customScoreArr
+				finalKeyArr.push(key)
+			end
+
+			finalKeyArr.reject { |item| item.nil? || item == '' }
+			csv << finalKeyArr
+		  
 			Judgement.find_each do |j|
-				@row = []
-				j.attributes.keys.each do |attr|
-					if attr == 'project_id'
-						@row << Project.find_by(id: j.attributes[attr]).title
-					elsif attr == 'user_id'
-						@row << User.find_by(id: j.attributes[attr]).full_name
-					else
-						@row << j.attributes[attr]
+			  
+				valuesArr = j.attributes.values
+				finalValueArr = Array.new
+				customValuesArr = j.attributes.values[6]
+
+				for i in 0..valuesArr.length
+					if i != 6
```

Comment:
Why is i!=6? You should replace this with something less cryptic.

Reply:
basically in the array, the 6th value is the custom fields which I want to avoid(because I add that later), I could instead check if the the array value is a json and then avoid it that way.